synapse/rest/client/models.py

```diff
@@ -36,18 +36,20 @@ class Config:
     type: Optional[StrictStr] = None
 
 
-class ThreePidRequestTokenBody(RequestBodyModel):
-    if TYPE_CHECKING:
-        client_secret: StrictStr
-    else:
-        # See also assert_valid_client_secret()
-        client_secret: constr(
-            regex="[0-9a-zA-Z.=_-]",  # noqa: F722
-            min_length=0,
-            max_length=255,
-            strict=True,
-        )
+if TYPE_CHECKING:
+    ClientSecretType = StrictStr
```

Comment:
tiny tiny question: would calling this `ClientSecretStr` make more sense so that the class is more readable without looking up the definition of this? I don't think it matters either way but it may be a good convention to have, in which case we would better consider that now

Reply:
Good idea, I like it. So what's the rule of thumb: if you need an alias for a `constr`, `conint` or `confloat`, call it `SomethingSomethingStr`, `FooBarInt` or `PotatoFloat` rather than `BlahBlahType`?

Perhaps you'd also like to see this for `conlist` (JSON arrays) and certain mapping types too?

Reply:
yeah, sounds good to me; I wouldn't do it for pydantic models though, so you wouldn't call e.g. `PostBody` `PostBodyMap`.